temporalcli/commands.batch_test.go

```diff
@@ -0,0 +1,231 @@
+					"--namespace", "batch-empty")
+				s.NoError(res.Err)
+				s.Empty(res.Stderr.String())
+				return strings.Count(res.Stdout.String(), "Completed") == 3
+			}, 5*time.Second, 100*time.Millisecond)
+
+			out := res.Stdout.String()
+			s.Equal(4, strings.Count(out, "\n"), "expect 3 data rows + 1 header row")
+			s.ContainsOnSameLine(out, "JobId", "State", "StartTime", "CloseTime") // header
+			s.ContainsOnSameLine(out, "TestBatchJob_List_2", "Completed")
+			s.ContainsOnSameLine(out, "TestBatchJob_List_1", "Completed")
+			s.ContainsOnSameLine(out, "TestBatchJob_List_0", "Completed")
+		})
+
+		t.Run("as text with limit", func(t *testing.T) {
+			res := s.Execute(
+				"batch", "list",
+				"--address", s.Address(),
+				"--namespace", "batch-empty",
+				"--limit", "1")
+			s.NoError(res.Err)
+			s.Empty(res.Stderr.String())
+
+			out := res.Stdout.String()
+			s.Equal(2, strings.Count(out, "\n"), "expect 1 data row + 1 header row")
+			s.ContainsOnSameLine(out, "JobId", "State", "StartTime", "CloseTime") // header
+		})
+
+		t.Run("as json", func(t *testing.T) {
+			res := s.Execute(
+				"batch", "list",
+				"--address", s.Address(),
+				"--namespace", "batch-empty",
+				"-o", "json")
+			s.NoError(res.Err)
+			s.Empty(res.Stderr.String())
+
+			out := res.Stdout.String()
+			s.ContainsOnSameLine(out, "\"jobId\": \"TestBatchJob_List_2\"")
+			s.ContainsOnSameLine(out, "\"jobId\": \"TestBatchJob_List_1\"")
+			s.ContainsOnSameLine(out, "\"jobId\": \"TestBatchJob_List_0\"")
+		})
+	})
+}
+
+func (s *SharedServerSuite) TestBatchJob_Terminate() {
+	s.t.Run("non-existing job id", func(t *testing.T) {
+		t.Run("as text", func(t *testing.T) {
+			res := s.Execute(
+				"batch", "terminate",
+				"--address", s.Address(),
+				"--job-id", "not-found",
+				"--reason", "testing")
+			s.Error(res.Err, errors.New("could not find Batch Job 'not-found'"))
+		})
+
+		t.Run("as json", func(t *testing.T) {
+			res := s.Execute(
+				"batch", "terminate",
+				"--address", s.Address(),
+				"--job-id", "not-found",
+				"--reason", "testing",
+				"-o", "json")
+			s.Error(res.Err, errors.New("could not find Batch Job 'not-found'"))
+		})
+	})
+
+	s.t.Run("existing job id", func(t *testing.T) {
+		t.Run("as text", func(t *testing.T) {
+			jobId := "TestBatchJob_Terminate_Text"
+			s.startBatchJob(jobId, s.Namespace())
+
+			res := s.Execute(
+				"batch", "terminate",
+				"--address", s.Address(),
+				"--job-id", jobId,
+				"--reason", "testing")
+			s.NoError(res.Err)
+			s.Empty(res.Stderr.String())
+			s.Equal("terminated Batch Job '"+jobId+"'\n", res.Stdout.String())
+		})
+
+		t.Run("as json", func(t *testing.T) {
+			jobId := "TestBatchJob_Terminate_JSON"
+			s.startBatchJob(jobId, s.Namespace())
+
+			res := s.Execute(
+				"batch", "terminate",
+				"--address", s.Address(),
+				"--job-id", jobId,
+				"--reason", "testing",
+				"-o", "json")
+			s.NoError(res.Err)
+			s.Empty(res.Stderr.String())
+			s.Empty(res.Stdout.String())
```

Comment:
I wasn't sure what to do here; the plain output shows a message; but I can't do that for JSON. The Server API response is empty.